engine/src/main/java/org/pentaho/di/trans/steps/setvaluefield/SetValueFieldMeta.java

```diff
@@ -48,10 +50,14 @@
 import org.pentaho.metastore.api.IMetaStore;
 import org.w3c.dom.Node;
 
+@InjectionSupported( localizationPrefix = "SetValueField.Injection.", groups = { "FIELDS" } )
 public class SetValueFieldMeta extends BaseStepMeta implements StepMetaInterface {
   private static Class<?> PKG = SetValueFieldMeta.class; // for i18n purposes, needed by Translator2!!
 
+  @Injection( name = "FIELD_NAME", group = "FIELDS" )
   private String[] fieldName;
+
+  @Injection( name = "REPLACE_BY_FIELD_VALUE", group = "FIELDS" )
```

Comment:
Out of curiosity - have you tested the injections without specifying this field in the injector? The reason I ask is because with the step I'm working with (Modified Java Script Value) I initially was getting ArrayOutOfBoundsExceptions, unless I provided values for ALL field columns, because these field related String arrays were not being sized properly (allocate method was not called) when going through MDI. I had to add provisions specifically to prevent these ArrayOutOfBoundsExceptions. This may not apply in your case, but just wanted to point this out.

Reply:
It appears it does happen with mine, nice catch. I'll look into adding handling for it, thanks.